src/fidesops/ops/schemas/connection_configuration/connection_secrets_saas.py

```diff
@@ -48,21 +76,34 @@ class SaaSSchemaFactory:
     def __init__(self, saas_config: SaaSConfig):
         self.saas_config = saas_config
 
-    # Pydantic uses the shorthand of (str, ...) to denote a required field of type str
+    # Pydantic uses the shorthand of (type, ...) to denote a required field of the given type
     def get_saas_schema(self) -> Type[SaaSSchema]:
         """Returns the schema for the current configuration"""
         field_definitions: Dict[str, Any] = {}
         for connector_param in self.saas_config.connector_params:
+            param_type = list if connector_param.multiselect else str
             field_definitions[connector_param.name] = (
                 connector_param.default_value
                 if connector_param.default_value
-                else (str, ...)
+                else (param_type, ...)
             )
         SaaSSchema.__doc__ = f"{str(self.saas_config.type).capitalize()} secrets schema"  # Dynamically override the docstring to create a description
+
+        # set the connector_params as a private attribute on the schema class
+        # so they can be accessible in the 'required_components_supplied' validator
         model: Type[SaaSSchema] = create_model(
             f"{self.saas_config.type}_schema",
             **field_definitions,
             __base__=SaaSSchema,
+            _connector_params=PrivateAttr(
```

Comment:
This was be best way I could find to dynamically set a property on the base class. I didn't include this with the other `field_definitions` since that is only used to describe the `secret` fields.